docs/index.md

```diff
@@ -86,24 +86,87 @@ connection "terraform" {
+  ]
+}
+```
+
+#### Configuring local file paths
+
+You can define a list of local directory paths to search for terraform files. Paths are resolved relative to the current working directory. For example:
+
+- `*.tf` matches all Terraform configuration files in the CWD.
+- `**/*.tf` matches all Terraform configuration files in the CWD and all sub-directories.
+- `../*.tf` matches all Terraform configuration files in the CWD's parent directory.
+- `steampipe*.tf` matches all Terraform configuration files starting with "steampipe" in the CWD.
+- `/path/to/dir/*.tf` matches all Terraform configuration files in a specific directory. For example:
+  - `~/*.tf` matches all Terraform configuration files in the home directory.
+  - `~/**/*.tf` matches all Terraform configuration files recursively in the home directory.
+- `/path/to/dir/main.tf` matches a specific file.
+
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [ "*.tf", "~/*.tf", "/path/to/dir/main.tf" ]
+}
+```
+
+**NOTE:** If paths includes `*`, all files (including non-Terraform configuration files) in the CWD will be matched, which may cause errors if incompatible file types exist.
+
+#### Configuring GitHub URLs
+
+You can define a list of URL as input to search for terraform files from a variety of protocols. For example:
+
+- `github.com/turbot/polygoat//*.tf` matches all top-level Terraform configuration files in the specified github repository.
+- `github.com/turbot/polygoat//**/*tf` matches all Terraform configuration files in the specified github repository and all sub-directories.
+- `github.com/turbot/polygoat?ref=fix_7677//**/*tf` matches all Terraform configuration files in the specific tag of github repository.
+- `git::https://github.com/turbot/steampipe-plugin-alicloud.git//alicloud-test/tests/alicloud_account//*.tf` matches all Terraform configuration files in the given HTTP URL using the Git protocol.
+
+If you want to download only a specific subdirectory from a downloaded directory, you can specify a subdirectory after a double-slash (`//`).
+
+- `github.com/turbot/polygoat//testing_frameworks/steampipe_mod_benchmark//*.tf` matches all Terraform configuration files in the specific folder of a github repo.
+
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [ "github.com/turbot/polygoat//*.tf", "github.com/turbot/polygoat//testing_frameworks/steampipe_mod_benchmark//*.tf", "git::https://github.com/turbot/steampipe-plugin-alicloud.git//alicloud-test/tests/alicloud_account//*.tf" ]
 }
 ```
 
-- `paths` - A list of directory paths to search for Terraform files. Paths are resolved relative to the current working directory. Paths may [include wildcards](https://pkg.go.dev/path/filepath#Match) and also support `**` for recursive matching. Defaults to the current working directory.
+#### Configuring S3 URLs
+
+You can also pass a S3 bucket URL to search all Terraform configuration files stored in the specified S3 bucket. For example:
```

Comment:
What permissions are used to access the S3 bucket URL? Do I need to configure the AWS CLI creds, or something else? Are only publicly readable buckets valid/readable with this feature?

Reply:
I tried to access a non-public bucket in `ap-northeast-2` with `paths = ["s3::https://s3-.amazonaws.com/bucket/my-bucket-name//**/*.tf"]` and received the following error:
```
cbruno@M1P ~ % steampipe query
Welcome to Steampipe v0.17.0-alpha.28
For more information, type .help
> select * from terraform_resource

Error: failed to get directory specified by the source s3::https://s3.amazonaws.com/bucket/my-bucket-name: error downloading 'https://s3.amazonaws.com/bucket/my-bucket-name': AccessDenied: Access Denied
	status code: 403, request id: WVPPEQDG2PTJCCKM, host id: jg2b0ZKlCD7aQo+wrgf3l624... (SQLSTATE HV000)

+------+------+-----------+-------+-----------+----------+------------+-----------+----------+------------+----------+--------+------+------+
| name | type | arguments | count | count_src | for_each | depends_on | lifecycle | provider | start_line | end_line | source | path | _ctx |
+------+------+-----------+-------+-----------+----------+------------+-----------+----------+------------+----------+--------+------+------+
+------+------+-----------+-------+-----------+----------+------------+-----------+----------+------------+----------+--------+------+------+

Time: 66ms.
```

I then tried adjusting paths to be `paths = ["s3::https://s3-ap-northeast-2.amazonaws.com/bucket/my-bucket-name//**/*.tf"]` based on https://github.com/hashicorp/go-getter#s3-bucket-examples, but got another error:
```
cbruno@M1P ~ % steampipe query
Welcome to Steampipe v0.17.0-alpha.28
For more information, type .help
> select * from terraform_resource

Error: failed to get directory specified by the source s3::https://s3-ap-northeast-2.amazonaws.com/bucket/my-bucket-name: error downloading 'https://s3-ap-northeast-2.amazonaws.com/bucket/my-bucket-name': BucketRegionError: incorrect region, the bucket is not in 'ap-northeast-2' region at endpoint '', bucket is in 'us-east-1' region
	status code: 301, request id: RHSD3M5M1V78WDRK, host id: zo8UUNiRzXXuEU/5/CQLSoZQpFOLWMsCaB1cyGDcaN... = (SQLSTATE HV000)

+------+------+-----------+-------+-----------+----------+------------+-----------+----------+------------+----------+--------+------+------+
| name | type | arguments | count | count_src | for_each | depends_on | lifecycle | provider | start_line | end_line | source | path | _ctx |
+------+------+-----------+-------+-----------+----------+------------+-----------+----------+------------+----------+--------+------+------+
+------+------+-----------+-------+-----------+----------+------------+-----------+----------+------------+----------+--------+------+------+

Time: 265ms.
```

I have my default AWS creds setup for this account, and confirmed I can access it through the AWS CLI:
```
cbruno@M1P ~ % aws s3api get-bucket-location --bucket my-bucket-name
{
    "LocationConstraint": "ap-northeast-2"
}
```

Reply:
@cbruno10 can you update your path to use the following?
```
paths = ["s3::https://s3.ap-northeast-2.amazonaws.com/bucket/my-bucket-name//**/*.tf"]
```